### ISSUE TITLE
Validate that Microsoft package content is also signed

### ISSUE THREAD

**Issue by karann-msft** [NuGet/NuGetGallery#6658]
This request came from @shahabhijeet

The ask is, NuGet.org should validate that the package contents are also signed.

**Comment by scottbommarito** [comment]
To clarify, you mean that NuGet.org should validate that the assemblies inside the package are signed?

**Comment by karann-msft** [comment]
not just assemblies, anything that can be signed. For e.g. .ps files

**Comment by shahabhijeet** [comment]
@karann-msft here are few things that I think you should consider when validating:
1. The publishing user of an organization should have a say what needs to be verified and what needs to be skipped.
2. There are few things that nuget should enforce and organization user cannot overwrite it:
 - for e.g. nuget package should be signed
 - certain nuget meta data should be available in certain format etc.
3. But organization user should have a say, whether nuget contents should be signed or not in case nuget.org is not enforcing that policy.
 - this is especially important within organization, where there can be multiple contributors who can have the ability to create publishing keys.
 - this is also crucial for us, if the owner of the nuget package is "Microsoft" and an organization within Microsoft like us.
 - Basically I am trying to find out a way where organization can provide a way to enforce policies on their nuget packages.

That being said, as v1 you should extend nuspec schema to allow organization to enforce validation policies, in this particular case, organization can specify if certain file types should be verified for authenticode signature.
This schema can then be uploaded to nuget to be associated with the organization account.
